hrjobcontract/js/test/controllers/modal/modal-contract.controller.spec.js

```diff
@@ -227,6 +227,23 @@ define([
       });
     });
 
+    describe('when closing the modal after removing the period end date', function () {
+      beforeEach(function () {
+        MockContract.contractEntity.details.period_end_date = moment().format('YYYY-MM-DD');
+        makeController();
+        $scope.entity.details.period_end_date = '';
+
+        $scope.save();
+        $scope.$digest();
+      });
+
+      it('returns a "have entitlement fields changed" variable set to true', function () {
```

Comment:
it either *sets* or *returns*. which one?

or you may make it "closes the modal and tells it that the entitlement fields have changed"